torch/_export/utils.py

```diff
@@ -862,6 +862,10 @@ def _extract_pytree_key(x):
         if node.op == "placeholder":
             assert node.name in name_map
             node.name = node.target = name_map[node.name]
+            # if the constant obj is an input, we also need to update meta["val"]
+            # because this is created before the placeholder naming pass
+            if isinstance(node.meta["val"], CustomObjArgument):
+                node.meta["val"].name = node.name
```

Comment:
@ydwu4 In https://github.com/pytorch/pytorch/pull/138749#discussion_r1813734529, we discussed about possibility of moving this change to the previous for loop, but it seems not that trivial as it would make more sense to combine everything into a one for loop if efficiency is a concern but keeping it like this seems easier to reason for me. What do you think?